app/static/js/volunteerProfile.js

```diff
@@ -0,0 +1,80 @@
+$(document).ready(function(){
+
+  $(".form-check-input").click(function updateInterest(){
+    var programID = $(this).attr('id');
+    var interest = $(this).is(':checked');
+    var username = $(this).attr('name');
+
+    if (interest) {
+      var routeUrl = "addInterest";
+    }
+    else {
+      var routeUrl = "deleteInterest";
+    }
+    interestUrl = "/" + routeUrl + "/" + programID + "/" + username;
+    $.ajax({
+      method: "POST",
+      url: interestUrl,
+      success: function(response) {
+          msgFlash("Your interest has been updated", "success");
+      },
+      error: function(request, status, error) {
+        console.log(status,error);
+        msgFlash("Error Updating Interest", "danger");
+        window.location.reload(true);
+      }
+    });
+  });
+
+  $(".ban").click(function() {
+
+    $("#banVolunteerButton").text($(this).val() + " Volunteer");
+    $(".modal-title").text($(this).val() + " Volunteer");
+    $("#modalProgramName").text("Program: " + $(this).attr("name"));
+    $("#banVolunteerModal").modal("toggle");
+    $("#banVolunteerButton").attr("programID", $(this).attr("id"))
+    $("#banVolunteerButton").attr("username", $(".form-check-input").attr("name"))
+    $("#banVolunteerButton").attr("banOrUnban", $(this).val());
+    $("#ubanEndDate").show()
+    $("#banVolunteerEndDate").val("")
+    $("#banVolunteerNote").val("")
+
+    if( $(this).val()=="Unban"){
+      $("#ubanEndDate").hide()
+      $("#banVolunteerEndDate").val("0001-01-01") //This is a placeholder value for the if statement in line 49 to work properly
+    }
+
+  });
+
+  $("#banVolunteerNote, #banVolunteerEndDate").change(function () {
+    if( ($("#banVolunteerNote").val()) && ($("#banVolunteerEndDate")).val()) {
+            $("#banVolunteerButton").prop("disabled", false);
+      }
+      else {
+        $("#banVolunteerButton").prop("disabled", true);
+      }

```

Comment:
Instead of switching on that boolean expression, you could just set the button to it:
```
$("#banVolunteerButton").prop("disabled", ! ($("#banVolunteerNote").val() && $("#banVolunteerEndDate").val()));
```
or to be clearer in two lines:
```
var enableButton = ($("#banVolunteerNote").val() && $("#banVolunteerEndDate").val());
$("#banVolunteerButton").prop("disabled", !enableButton);
```
Also, in this whole file you'd save yourself a lot of typing and line space if you just used 'ban' instead of 'banVolunteer' everywhere. The extra word doesn't really add to the clarity, and you use 'unban' instead of 'unbanVolunteer'

Reply:
Fixed!